File: packages/autonavi_location_flutter/android/src/main/kotlin/com/autonavi/location/flutter/AutonaviLocationPlugin.kt
package com.autonavi.location.flutter

import android.content.Context
import com.amap.api.location.AMapLocationClient
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class AutonaviLocationPlugin : FlutterPlugin, MethodChannel.MethodCallHandler {

    private lateinit var methodChannel: MethodChannel
    private lateinit var eventChannel: EventChannel
    private lateinit var geofenceEventChannel: EventChannel
    private var streamHandler: LocationStreamHandler? = null
    private var context: Context? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        context = binding.applicationContext

        // Privacy compliance
        AMapLocationClient.updatePrivacyShow(binding.applicationContext, true, true)
        AMapLocationClient.updatePrivacyAgree(binding.applicationContext, true)

        methodChannel = MethodChannel(
            binding.binaryMessenger,
            "plugins.autonavi.flutter/location_method"
        )
        methodChannel.setMethodCallHandler(this)

        streamHandler = LocationStreamHandler(binding.applicationContext)
        eventChannel = EventChannel(
            binding.binaryMessenger,
            "plugins.autonavi.flutter/location"
        )
        eventChannel.setStreamHandler(streamHandler)

        geofenceEventChannel = EventChannel(
            binding.binaryMessenger,
            "plugins.autonavi.flutter/geofence_events"
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        methodChannel.setMethodCallHandler(null)
        streamHandler?.dispose()
        streamHandler = null
        context = null
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        val ctx = context ?: run {
            result.error("NOT_INITIALIZED", "Plugin not initialized", null)
            return
        }

        when (call.method) {
            "location#getOnce" -> {
                @Suppress("UNCHECKED_CAST")
                val options = call.arguments as? Map<String, Any> ?: emptyMap()
                LocationStreamHandler(ctx).getOnce(
                    options,
                    onSuccess = { result.success(it) },
                    onError = { code, msg -> result.error(code, msg, null) },
                )
            }
            "location#updateOptions" -> {
                @Suppress("UNCHECKED_CAST")
                val options = call.arguments as? Map<String, Any> ?: emptyMap()
                streamHandler?.updateOptions(options)
                result.success(null)
            }
            "geofence#addCircle" -> {
                @Suppress("UNCHECKED_CAST")
                val args = call.arguments as? Map<String, Any> ?: emptyMap()
                // Geofence implementation would use AMapLocationClient's geofence API
                result.success(null)
            }
            "geofence#remove" -> {
                result.success(null)
            }
            "geofence#removeAll" -> {
                result.success(null)
            }
            else -> result.notImplemented()
        }
    }
}

File: packages/autonavi_location_flutter/ios/Classes/AutonaviLocationPlugin.swift
import Flutter
import AMapLocationKit

public class AutonaviLocationPlugin: NSObject, FlutterPlugin {

    private var methodChannel: FlutterMethodChannel?
    private var streamHandler: LocationStreamHandler?

    public static func register(with registrar: FlutterPluginRegistrar) {
        // Privacy compliance
        AMapLocationManager.updatePrivacyShow(.didShow, privacyInfo: .contain)
        AMapLocationManager.updatePrivacyAgree(.did)

        let instance = AutonaviLocationPlugin()

        let methodChannel = FlutterMethodChannel(
            name: "plugins.autonavi.flutter/location_method",
            binaryMessenger: registrar.messenger()
        )
        registrar.addMethodCallDelegate(instance, channel: methodChannel)
        instance.methodChannel = methodChannel

        let streamHandler = LocationStreamHandler()
        instance.streamHandler = streamHandler

        let eventChannel = FlutterEventChannel(
            name: "plugins.autonavi.flutter/location",
            binaryMessenger: registrar.messenger()
        )
        eventChannel.setStreamHandler(streamHandler)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        let args = call.arguments as? [String: Any] ?? [:]

        switch call.method {
        case "location#getOnce":
            streamHandler?.getOnce(options: args) { locationResult in
                result(locationResult)
            }
        case "location#updateOptions":
            streamHandler?.updateOptions(args)
            result(nil)
        case "geofence#addCircle",
             "geofence#remove",
             "geofence#removeAll":
            // Geofence implementation placeholder
            result(nil)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
